app/src/main/java/org/mozilla/fenix/settings/account/AccountSettingsFragment.kt

```diff
@@ -125,60 +152,54 @@ class AccountSettingsFragment : PreferenceFragmentCompat() {
         )
     }
 
+    private fun syncNow() {
+        lifecycleScope.launch {
+            requireComponents.analytics.metrics.track(Event.SyncAccountSyncNow)
+            // Trigger a sync.
+            requireComponents.backgroundServices.accountManager.syncNowAsync().await()
+            // Poll for device events.
+            accountManager.authenticatedAccount()
+                ?.deviceConstellation()
+                ?.refreshDeviceStateAsync()
+                ?.await()
+        }
+    }
+
+    private fun syncDeviceName(newValue: String): Boolean {
```

Comment:
@grigoryk Is it okay to await this on the main thread? 

@jyeontaek We might want to turn this into a method that can better handle the asynchronous code  using callbacks or a delegate 👍